app/actions.js

```diff
@@ -407,29 +405,73 @@ const initializeCabal = ({ addr, username, settings }) => async dispatch => {
   const isNew = !addr
   const cabalDetails = isNew ? await client.createCabal() : await client.addCabal(addr)
   addr = cabalDetails.key
-  client.focusCabal(addr)
 
-  let firstUpdate = true
-  cabalDetails.on('update', throttle((details) => {
-    const users = details.getUsers()
-    const username = details.getLocalName()
-    const channels = details.getChannels()
-    const channelsJoined = details.getJoinedChannels()
-    const channelMessagesUnread = getCabalUnreadMessagesCount(details)
-    const currentChannel = details.getCurrentChannel()
-    const channelMembers = details.getChannelMembers()
+  function onUpdate () {
+    const users = cabalDetails.getUsers()
+    const username = cabalDetails.getLocalName()
+    const channels = cabalDetails.getChannels()
+    const channelsJoined = cabalDetails.getJoinedChannels()
+    const channelMessagesUnread = getCabalUnreadMessagesCount(cabalDetails)
+    const currentChannel = cabalDetails.getCurrentChannel()
+    const channelMembers = cabalDetails.getChannelMembers()
     dispatch({ type: 'UPDATE_CABAL', addr, channelMessagesUnread, users, username, channels, channelsJoined, currentChannel, channelMembers })
     dispatch(getMessages({ addr, amount: 1000, channel: currentChannel }))
     dispatch(updateAllsChannelsUnreadCount({ addr, channelMessagesUnread }))
-    if (firstUpdate) {
-      firstUpdate = false
-      // Focus default or last channel viewed
-      dispatch(viewCabal({ addr, channel: settings.currentChannel }))
+  }
+  const onUpdateThrottled = throttle(onUpdate, 500)
+
+  const cabalDetailsEvents = [
+    {
+      name: 'cabal-focus',
+      action: (data) => {}
+    }, {
+      name: 'channel-focus',
+      action: onUpdateThrottled
+    }, {
+      name: 'channel-join',
+      action: onUpdateThrottled
+    }, {
+      name: 'channel-leave',
+      action: onUpdateThrottled
+    }, {
+      name: 'new-channel',
+      action: onUpdateThrottled
+    }, {
+      name: 'new-message',
+      action: onUpdateThrottled
+    }, {
+      name: 'publish-message',
+      action: onUpdateThrottled
+    }, {
+      name: 'publish-nick',
+      action: onUpdateThrottled
+    }, {
+      name: 'started-peering',
+      action: onUpdateThrottled
+    }, {
+      name: 'status-message',
+      action: onUpdateThrottled
+    }, {
+      name: 'stopped-peering',
+      action: onUpdateThrottled
+    }, {
+      name: 'topic',
+      action: onUpdateThrottled
+    }, {
+      name: 'user-updated',
+      action: onUpdateThrottled
```

Comment:
i don't see how this is any different than what existed before. how/where is the loop you're talking about get terminated?

Reply:
also; if you use `DEBUG="cabal-client" <run desktop>` you should be able to see which events are being fired tons of times.

my recommendation is, if it's something easily handled like `start-peering`/`stopped-peering` is then to write a separate handler for that event, rather than reuse `onUpdateThrottled` as it should cost way less cpu / cause less sluggishness

Reply:
yeah really it was the `cabal-focus` event was triggering Desktop to trigger another update which make another `cabal-focus` event and loop. i didnt split the logic up of what each event should do so i could get this out the door asap but i can!

Reply:
aha! i'm so glad that disambiguating the events made it possible for you to find that! and so easily :D! 

if that's the solution and the sluggishness is gone, then i say SHIP IT 📦

Reply:
woo! just pushed with the logic split between the events! now no throttling at all!